ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/view/ToolAdminViewImpl.java

```diff
@@ -0,0 +1,292 @@
+                                                                       appearance.descriptionColumnLabel());
+        ColumnConfig<Tool, String> locationCol = new ColumnConfig<>(toolProps.location(),
+                                                                    appearance.locationColumnInfoWidth(),
+                                                                    appearance.locationColumnInfoLabel());
+        ColumnConfig<Tool, String> typeCol = new ColumnConfig<>(toolProps.type(),
+                                                                appearance.typeColumnInfoWidth(),
+                                                                appearance.typeColumnInfoLabel());
+        ColumnConfig<Tool, String> attributionCol = new ColumnConfig<>(toolProps.attribution(),
+                                                                       appearance.attributionColumnWidth(),
+                                                                       appearance.attributionColumnLabel());
+        ColumnConfig<Tool, String> versionCol = new ColumnConfig<>(toolProps.version(),
+                                                                   appearance.versionColumnInfoWidth(),
+                                                                   appearance.versionColumnInfoLabel());
+
+        list.add(nameCol);
+        list.add(descriptionCol);
+        list.add(locationCol);
+        list.add(typeCol);
+        list.add(attributionCol);
+        list.add(versionCol);
+        return new ColumnModel<>(list);
+    }
+
+    @Override
+    public void setPresenter(Presenter presenter) {
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void setToolList(List<Tool> tools) {
+        listStore.replaceAll(tools);
+    }
+
+    @Override
+    public void setToolDetails(final Tool tool) {
+        final ToolAdminDetailsWindow detailsPanel = ToolAdminDetailsWindow.addToolDetails(factory);
+        detailsPanel.edit(tool);
+
+        final IPlantDialog dialogWindow = getIplantDialogWindow();
+
+        final TextButton delete = getDeleteButton(tool, dialogWindow);
+        dialogWindow.addButton(delete);
+        dialogWindow.getOkButton().setText(appearance.dialogWindowUpdateBtnText());
+        dialogWindow.addOkButtonSelectHandler(new SelectEvent.SelectHandler() {
+            @Override
+            public void onSelect(SelectEvent event) {
+                Tool tool = detailsPanel.getTool();
+                if (detailsPanel.isValid()){
+                    presenter.updateTool(tool);
+                    dialogWindow.hide();
+                    try {
+                        grid.getSelectionModel().deselect(grid.getSelectionModel().getSelectedItem());
+                    } catch (UmbrellaException umb) {
+                        //do nothing -- throws error about DeviceList, VolumeList, and VolumesFromList having null IDs
```

Comment:
There should be a better way of handling this than catching an `UmbrellaException`.
Since these IDs are not required in the add/update requests, maybe they can be removed from the UI's model, and the grid's listStore model-keys can be managed with AutoBean tags instead?